src/atomate2/vasp/jobs/defect.py

```diff
@@ -3,27 +3,373 @@
+    sc_mat : NDArray | None
+        The supercell matrix. if None, the code will attempt to create a
+        nearly-cubic supercell.
+
+    Returns
+    -------
+    Response:
+        Output a dictionary containing the bulk supercell calculation summary.
+    """
+    _logger.info("Running bulk supercell calculation. Running...")
+    sc_mat = get_sc_fromstruct(uc_structure) if sc_mat is None else sc_mat
+    sc_mat = np.array(sc_mat)
+    sc_structure = uc_structure * sc_mat
+    relax_job = relax_maker.make(sc_structure)
+    relax_job.name = "bulk relax"
+    relax_output: TaskDocument = relax_job.output
+
+    summary_d = dict(
+        uc_structure=uc_structure,
+        sc_entry=relax_output.entry,
+        sc_struct=relax_output.structure,
+        sc_mat=sc_mat.tolist(),
+        dir_name=relax_output.dir_name,
+        uuid=relax_job.uuid,
+    )
+    # waiter = wait_for_dict(summary_d)
+    return Response(output=summary_d, replace=[relax_job])
+
+
+@job
+def spawn_defect_calcs(
+    defect_gen: DefectGenerator,
+    sc_mat: NDArray,
+    relax_maker: RelaxMaker,
+) -> Response:
+    """Spawn defect calculations from the DefectGenerator.
+
+    Parameters
+    ----------
+    defect_gen : DefectGenerator
+        The defect generator to use.
+    sc_mat : NDArray
+        The supercell matrix. If None, the code will attempt to create a
+        nearly-cubic supercell.
+
+    Returns
+    -------
+    Response:
+        The response containing the outputs of the defect calculations as a dictionary
+    """
+    defect_q_jobs = []
+    output = dict()
+    name_counter: dict = defaultdict(lambda: 0)
+
+    for defect in defect_gen:
+        q_job = run_all_charge_states(
+            defect,
+            sc_mat=sc_mat,
+            relax_maker=relax_maker,
+            defect_index=f"{name_counter[defect.name]}",
+        )
+        defect_q_jobs.append(q_job)
+        output[f"{defect.name}-{name_counter[defect.name]}"] = dict(
+            defect=defect,
+            results=q_job.output,
+        )
+        name_counter[defect.name] += 1
+    return Response(output=output, replace=defect_q_jobs)
 
-logger = logging.getLogger(__name__)
 
+@job
+def run_all_charge_states(
+    defect: Defect,
+    relax_maker: RelaxMaker | None = None,
+    sc_mat: NDArray | None = None,
+    defect_index: str = "",
+    add_info: dict | None = None,
+) -> Response:
+    """Perform charge defect supercell calculations and save the Hartree potential.
+
+    Run a ISIF2 calculation for each available charge state of the defect.
+    Ensure that the LOCPOT file is stored in the output.
+
+    Parameters
+    ----------
+    defect:
+        A defect object representing the defect in a unit cell.
+    relax_maker:
+        A RelaxMaker object to use for the atomic relaxation.
+        If None, the default will be used (see DEFAULT_RELAX_MAKER).
+    sc_mat:
+        The supercell matrix. If None, the code will attempt to create a
```

Comment:
This is now changed right?

Reply:
technically this function did not need to change since the `sc_mat` will always be supplied from a previous calculation.
The default `sc_mat = None` behavior is just the default behavior of the defects packages.